plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/extensions/FunctionCallTransformer.kt

```diff
@@ -421,6 +429,110 @@ class FunctionCallTransformer(
         return newCall1
     }
 
+    private fun materialize(
+        dataFrameSchema: PluginDataFrameSchema,
```

Comment:
I can't say I have any idea what's going on in this massive file. Might it be an idea to split it up eventually? I can't review it if I don't understand it.

Reply:
as a starting point i recommend to read kdocs for FirFunctionCallRefinementExtension